week-2/Homework/mandatory/1-fix-functions.js

```diff
@@ -12,9 +12,14 @@ function mood() {
 }
 
 function greaterThan10() {
-  let num = 10;
-  let isBigEnough;
-
+  let num = 12;
+  let isBigEnough = function() {
+    if (num > 10) {
+      return true;
+    } else {
+      return false;
+    }
+  };
   if (isBigEnough) {
```

Comment:
This is really interesting! `isBigEnough` here is the actual function, not the value of calling the function (because it isn't `isBigEnough()`) - so what we've learnt here is that like how `0` is treated like `false` in an `if` condition, and `1` is treated like `true`, a function is also treated like `true`. So here, this condition will always evaluate to `true`, not matter what value `num` has.

You probably wanted to put:
```javascript
if (isBigEnough()) {
```
but a little simpler than that would be to skip the function completely; you can even write:
```javascript
let isBigEnough = num > 10;
```
and that should work fine :)

Reply:
Thank you. I am correcting it now,but just added () and did not go to the final nicer way because of limited time. I still need to finish my 4th exercise.Almost done but my bubble sort is not working and i figure out how to fix it i will return and try the third way:)